burnman/layer.py

```diff
@@ -0,0 +1,797 @@
+from __future__ import print_function
+# This file is part of BurnMan - a thermoelastic and thermodynamic toolkit for the Earth and Planetary Sciences
+# Copyright (C) 2012 - 2017 by the BurnMan team, released under the GNU
+# GPL v2 or later.
+
+import numpy as np
+from scipy.integrate import odeint
+from scipy.integrate import quad
+from scipy.interpolate import UnivariateSpline
+from burnman import averaging_schemes
+from burnman import constants
+from burnman import geotherm
+import warnings
+
+
+from .material import Material, material_property
+from .mineral import Mineral
+from .composite import Composite
+from .seismic import Seismic1DModel
+
+import matplotlib.pyplot as plt
+
+
+class Layer(object):
+    """
+    A planetary layer class
```

Comment:
This is going to be the main summary in the autogenerated docs. It would be good to describe in more detail what this class is, and what it does.